examples/mobile/src/lib.rs

```diff
@@ -152,7 +152,9 @@ fn button_handler(
     }
 }
 
-fn setup_music(asset_server: Res<AssetServer>, audio: Res<Audio>) {
-    let music = asset_server.load("sounds/Windless Slopes.ogg");
-    audio.play(music);
+fn setup_music(asset_server: Res<AssetServer>, mut commands: Commands) {
+    commands.spawn(
+        AudioBundle::from_audio(asset_server.load("sounds/Windless Slopes.ogg"))
```

Comment:
It would be nice to be able to do this here:

```rust
commands.spawn(AudioBundle {
    source: asset_server.load("sounds/Windless Slopes.ogg"),
    settings: PlaybackSettings::LOOP,
});
```

It involves less typing / syntax, it has a simpler mental model (type + fields instead of type + fields you don't set + functions that initialize them), and it is generally more Bevy-idiomatic for Bundle init.

There are also "redundant" functions like `AudioBundle::from_audio` and `AudioBundle::from_custom_audio`. It would be nice if we could just do a standard `From` impl like `AudioBundle::from`.

That being said, I understand why we're doing it this way currently. Due to the way Rust's type inference works, even though we're specifying `AudioBundle<Source = AudioSource>`, it cannot infer the type of the `source: Handle<Source>` field in cases like `asset_server.load()`. In practice Rust's "default generic concrete types" are frustratingly (and confusingly) limited.

That being said, we've solved this problem elsewhere in the engine by replacing "default generic concrete types" with type aliases (ex:  `pub type PbrBundle = MaterialMeshBundle<StandardMaterial>`)

This lets us use generics freely without the need to define "special concrete path functions" or avoid using struct-init-syntax. Everything "just works" as expected.

I vote that we rename `AudioBundle<Source>` to something like `AudioSourceBundle<Source>` and then create a `pub type AudioBundle =  AudioSourceBundle<AudioSource>`.